testing/hierarchies.py

```diff
@@ -990,10 +990,14 @@ def _compare_children(hierarchy_type, test, test_child, exhale_child):
                     exhale_grand_child = candidate
                     break
         else:
+            identifier = test_grand_child.breathe_identifier()
             for candidate in exhale_child.children:
-                if candidate.kind == test_grand_child.kind and candidate.name == test_grand_child.name:
+                if candidate.kind == test_grand_child.kind and \
+                        candidate.breathe_identifier() == identifier:
                     exhale_grand_child = candidate
         if not exhale_grand_child:
+            import ipdb
+            ipdb.set_trace()
```

Comment:
`tox -e py -- -k cpp_nesting -s`, `-k` to select this module and `-s` to get the debugger active.  otherwise comment this out and test fail.

FYI one of the tests generates HTML in this module, so if you do `-k cpp_nesting` then afterward the file `testing/projects/cpp_nesting/docs_CPPNestingPages_test_html_output/_build/html/index.html` can be viewed.